tests/integration/test_dq_rules.py

```diff
@@ -0,0 +1,153 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+
+import logging
+import shutil
+
+from google.cloud import bigquery
+
+import pytest
+
+from clouddq.integration.bigquery.bigquery_client import BigQueryClient
+from clouddq.main import main
+from clouddq.runners.dbt.dbt_runner import DbtRunner
+from clouddq.runners.dbt.dbt_utils import get_dbt_invocation_id
+from clouddq.utils import working_directory
+
+
+logger = logging.getLogger(__name__)
+
+class TestDqRules:
+
+    @pytest.fixture(scope="session")
+    def client(self):
+        """Get BigQuery Client using discovered ADC"""
+        client = BigQueryClient()
+        yield client
+        client.close_connection()
+
+    @pytest.fixture(scope="session")
+    def create_expected_results_table(
+        self,
+        client,
+        test_resources,
+        gcp_project_id,
+        target_bq_result_dataset_name,
+        target_bq_result_table_name,):
+
+        client = client.get_connection()
+
+        table_id = f"{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}_expected"
+
+        job_config = bigquery.LoadJobConfig(
+            source_format=bigquery.SourceFormat.CSV,
+            skip_leading_rows=1,
+            autodetect=True,
+            write_disposition="WRITE_TRUNCATE")
+
+        with open(test_resources / "expected_results.csv", "rb") as source_file:
+            job = client.load_table_from_file(source_file, table_id, job_config=job_config)
+
+        job.result()  # Waits for the job to complete.
+
+        table = client.get_table(table_id)  # Make an API request.
+        print(f"Loaded {table.num_rows} rows and {len(table.schema)} columns to {table_id}")
+
+    def test_dq_rules(
+        self,
+        runner,
+        temp_configs_dir,
+        gcp_application_credentials,
+        gcp_project_id,
+        gcp_bq_dataset,
+        gcp_bq_region,
+        target_bq_result_dataset_name,
+        target_bq_result_table_name,
+        tmp_path,
+        client,
+        gcp_impersonation_credentials,
+        gcp_sa_key,
+        create_expected_results_table,
+    ):
+
+        try:
+            temp_dir = Path(tmp_path).joinpath("clouddq_test_cli_integration_4")
+            temp_dir.mkdir(parents=True)
+            with working_directory(temp_dir):
+                logger.info(f"test_last_modified_in_dq_summary {gcp_application_credentials}")
+                target_table = f"{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}"
+                args = [
+                    "T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE",
+                    f"{temp_configs_dir}",
```

Comment:
can we make explicit which config while was used in this test? Ideally, this config file should be created as a part of this test so that if anything fails we can easily figure out which YAML file was used for the test.

Ideally it is a single YAML file being tested rather than a directory & we should print out the YAML file as part of the test so we can inspect it immediately in case of test failures.